File: src/main/java/sprykerFileCreator/model/definition/spryker/ClientFactoryDefinition.java
package pav.sprykerFileCreator.model.definition.spryker;

import pav.sprykerFileCreator.model.definition.ClassDefinitionInterface;
import pav.sprykerFileCreator.model.generator.SprykerConstants;

public class ClientFactoryDefinition implements ClassDefinitionInterface {

    @Override
    public String getClassType() {
        return SprykerConstants.CLIENT_FACTORY;
    }

    @Override
    public String getNamePattern() {
        return "{{ bundleName }}Factory";
    }

    @Override
    public String getNamespacePattern() {
        return "{{ projectName }}\\Client\\{{ bundleName }}";
    }

    @Override
    public String getMethodForReturnType() {
        return "getFactory()";
    }

    @Override
    public String[] getDocBlockClasses() {
        String [] docBlockClasses = {};

        return docBlockClasses;
    }

    @Override
    public String getDefaultParentClass() {
        return "Spryker\\Client\\Kernel\\AbstractFactory";
    }

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/services/definitions/spryker/ClientFactoryDefinition.kt
package com.github.tobi812.sprykerplugin.services.definitions.spryker

import com.github.tobi812.sprykerplugin.constants.SprykerConstants
import com.github.tobi812.sprykerplugin.services.definitions.ClassDefinitionInterface

class ClientFactoryDefinition : ClassDefinitionInterface {
    override val classType: String
        get() = SprykerConstants.CLIENT_FACTORY
    override val namePattern: String
        get() = "{{ bundleName }}Factory"
    override val namespacePattern: String
        get() = "{{ projectName }}\\Client\\{{ bundleName }}"
    override val methodForReturnType: String
        get() = "getFactory()"
    override val docBlockClasses: Array<String>
        get() = arrayOf()
    override val defaultParentClass: String
        get() = "Spryker\\Client\\Kernel\\AbstractFactory"
}
